mm2src/adex_cli/src/scenarios/init_mm2_cfg.rs

```diff
@@ -0,0 +1,331 @@
+            rpc_password: None,
+            seed_phrase: None,
+            allow_weak_password: None,
+            dbdir: None,
+            rpcip: None,
+            rpcport: None,
+            rpc_local_only: None,
+            i_am_seed: None,
+            seednodes: Vec::<Ipv4Addr>::new(),
+            hd_account_id: None,
+        }
+    }
+
+    fn inquire(&mut self) -> Result<(), ()> {
+        self.inquire_gui()?;
+        self.inquire_net_id()?;
+        self.inquire_seed_phrase()?;
+        self.inquire_allow_weak_password()?;
+        self.inquire_rpc_password()?;
+        self.inquire_dbdir()?;
+        self.inquire_rpcip()?;
+        self.inquire_rpcport()?;
+        self.inquire_rpc_local_only()?;
+        self.inquire_i_am_a_seed()?;
+        self.inquire_seednodes()?;
+        self.inquire_hd_account_id()?;
+        Ok(())
+    }
+
+    #[inline]
+    fn inquire_dbdir(&mut self) -> Result<(), ()> {
+        let is_reachable_dir = |dbdir: &InquireOption<String>| -> Result<Validation, CustomUserError> {
+            match dbdir {
+                InquireOption::None => Ok(Validation::Valid),
+                InquireOption::Some(dbdir) => {
+                    let path = Path::new(dbdir);
+                    if path.is_dir().not() {
+                        return Ok(Validation::Invalid(
+                            format!("\"{dbdir}\" - is not a directory or does not exist").into(),
+                        ));
+                    }
+                    Ok(Validation::Valid)
+                },
+            }
+        };
+
+        self.dbdir = CustomType::<InquireOption<String>>::new("What is dbdir")
+                .with_placeholder(DEFAULT_OPTION_PLACEHOLDER)
+                .with_help_message("AtomicDEX API database path. Optional, defaults to a subfolder named DB in the path of your mm2 binary")
+                .with_validator(is_reachable_dir)
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get dbdir: {error}");
+                })?.into();
+
+        Ok(())
+    }
+
+    #[inline]
+    fn inquire_gui(&mut self) -> Result<(), ()> {
+        self.gui = Some(DEFAULT_GID.into());
+        info!("> gui is set by default: {DEFAULT_GID}");
+        Ok(())
+    }
+
+    #[inline]
+    fn inquire_net_id(&mut self) -> Result<(), ()> {
+        self.net_id = CustomType::<u16>::new("What is the network `mm2` is going to be a part, net_id:")
+                .with_default(DEFAULT_NET_ID)
+                .with_help_message(r#"Network ID number, telling the AtomicDEX API which network to join. 7777 is the current main network, though alternative netids can be used for testing or "private" trades"#)
+                .with_placeholder(format!("{DEFAULT_NET_ID}").as_str())
+                .prompt()
+                .map_err(|error| {
+                    error!("Failed to get net_id: {error}");
+                })?.into();
+        Ok(())
+    }
+
+    #[inline]
```

Comment:
I see you're #[inline] most of this functions which I'll disagree as I think only highly performance critical functions should be #[inline] in most cases as inline also adds to compilation time.. but I'm wrong 🙄

Reply:
[ref](https://github.com/KomodoPlatform/atomicDEX-API/pull/1729#discussion_r1149689550)

Reply:
hmm as I remember we had topics about inline several times https://github.com/KomodoPlatform/atomicDEX-API/pull/1636#discussion_r1106917518
https://matklad.github.io/2021/07/09/inline-in-rust.html#Inlining-in-Rust
may be use inline always here too?